### ISSUE TITLE
Function for Morphological analyzer

### ISSUE THREAD

**Issue by sinaahmadi** [MSeal/cython_hunspell#21]
Hi,

Thanks for this interesting tool.

I am wondering if there is a way to also use the Hunspell morphologial analyzer, something similar to the `analyze` command.

Thanks.

**Comment by MSeal** [comment]
I did use this library for an early prototype of a morphological analyzer many years ago by creating a custom dictionary to capture the stems and other cousin relations of all words in a corpus, and then doing a suggestion search across them. The Hunspell library can pack a surprising number of elements and still perform. I ended up deviating from that as the state space I was searching got so large that I needed first and second order buckets to organize terms to compare. And after that I was doing more complicated analysis that didn't all fit inside the manipulation search space, so I can't speak to how it worked after that point, but if you were doing simple analysis with a know pre-processing algorithm, it should work for that in extended use cases as well.

**Comment by sinaahmadi** [comment]
Thanks @MSeal for your clarification.

I have an operational morphological analyzer and am thinking of a way to integrate it in my Python modules. Thanks to your efforts, I am able to use the spell-checking functionalities, but not morphological analysis. 

Do you think it'll be possible to include the `analyze` function in the wrapper? Or, is there any workaround to do so?

**Comment by MSeal** [comment]
Yeah I think that'd be reasonable to add. Let me see if I can find a little time this weekend to expose that.

**Comment by sinaahmadi** [comment]
Thanks @MSeal. That will be wonderful!

**Comment by MSeal** [comment]
I've been fighting builds all weekend. I have the new functionality but I needed to revamp the builds and that proved to be a monumental effort. I'll keep working on fixing the build to unblock a 2.0.0 release once I can get travis to create the binaries correct on each platform.

**Comment by sinaahmadi** [comment]
Thank you very much @MSeal . Your efforts are so appreciated.
I reckon you'd have testing material yourself. If no, I can provide you with parts of my annotated lexicon (`.dic` file containing flags) and morphological rules (`.aff` file).

**Comment by MSeal** [comment]
@sinaahmadi The new 2.0.0 release has all the new hunspell functions except `generate` available now. README is updated with how to use those functions. See if that works for you :)

**Comment by sinaahmadi** [comment]
@MSeal Thanks a million. It already looks very promising.

On another note, I am getting `ModuleNotFoundError: No module named 'hunspell'` error after upgrading the package. Do you think there would be an issue that explains it? I checked my packages and I have `cyhunspell` installed.

**Comment by MSeal** [comment]
Hmm I'll have to get a VM setup to check what's going on for Mac. I have a Linux and Windows machine to test locally but can only rely on the Travis tests for Mac. Can you double check that `python -m site` / `python -c 'import site; print(site.getsitepackages())'` prints out the path / packages you expect where cyhunspell should be installed?

**Comment by MSeal** [comment]
Ok yeah I can see that the wheel that got published was missing the binary that got built. I'll look into what went wrong with the travis build later today.

**Comment by MSeal** [comment]
I'm having trouble getting the mac build to work correctly in travis, though I can get it working just fine locally. I'll keep playing with travis changes but if you want to use in the meantime you should be able to download it and build locally with (it'll take a bit to finish):

```
pip install git+https://github.com/MSeal/cython_hunspell@2.0.0
```

**Comment by sinaahmadi** [comment]
Thanks @MSeal for following up on this issue.

I am unfortunately not able to even use it when I build it locally. However, I re-installed the previous version (`1.3.4') and it works just fine as before. So, I am sure there is no issue with my Python package management on my Mac.

**Comment by sinaahmadi** [comment]
Dear @MSeal ,
Have you had time to look into this?

**Comment by MSeal** [comment]
Yes but I haven't gotten a fix yet for mac builds (I have to wait 15 minutes between each change to see what the build logs spit out :/). I'll try some more things this evening.

**Comment by MSeal** [comment]
Sorry this is till unfixed, work has kept me very busy.

**Comment by sinaahmadi** [comment]
Dear Matthew,
Would you please let me know if you'd have any update with this regard?

**Comment by MSeal** [comment]
Apologies, I gave up in frustration on this issue for a bit. But I think I have a test build finally passing with a proper wheel for osx as of today. I should have it out later tonight if testing it goes smoothly.

**Comment by MSeal** [comment]
Gah there's another issue with osx -- 2.0.1a0 has an almost working version :/ (testing on a mac vm this time around)

**Comment by MSeal** [comment]
Should have that additional issue fixed by tomorrow (have to wait for half-hour builds to test each change for this part)

**Comment by sinaahmadi** [comment]
Thanks so much @MSeal. Looking impatiently forward to seeing this 👍

**Comment by MSeal** [comment]
@sinaahmadi 2.0.1 is working in my OSX VM now. Future updates should be a lot quicker now that the build path works for each OS.
